models/models/output_file.py

```diff
@@ -0,0 +1,216 @@
+            meta=kwargs.get('meta'),
+            valid=parse_sql_bool(kwargs.get('valid')),
+            secondary_files=kwargs.get('secondary_files', {}),
+        )
+
+    def to_external(self):
+        """
+        Convert to external model
+        """
+        return OutputFile(
+            id=self.id,
+            parent_id=self.parent_id,
+            path=self.path,
+            basename=self.basename,
+            dirname=self.dirname,
+            nameroot=self.nameroot,
+            nameext=self.nameext,
+            file_checksum=self.file_checksum,
+            size=self.size,
+            meta=self.meta,
+            valid=self.valid,
+        )
+
+    @staticmethod
+    async def get_file_info(
+        file_obj: CloudPath | AnyPath | GSPath | Path, client: Client
+    ) -> dict | None:
+        """Get file info for file at given path"""
+        try:
+            file_checksum = None
+            valid = False
+            size = 0
+            # try:
+            if isinstance(file_obj, GSPath) and client:
+                if '/' in file_obj.blob:
+                    prefix = '/'.join(file_obj.blob.split('/')[:-1]) + '/'
+                else:
+                    prefix = ''
+                delimiter = '/'
+                blobs = client.list_blobs(
+                    bucket_or_name=str(file_obj.bucket),
+                    versions=False,
+                    prefix=prefix,
+                    delimiter=str(delimiter),
+                )  # pylint: disable=E1101
+                for blob in blobs:
+                    if blob.name == file_obj.blob:
+                        if file_obj.suffix != '.mt':
+                            file_checksum = blob.crc32c  # pylint: disable=E1101
+                            valid = True
+                            size = blob.size  # pylint: disable=E1101
+
+            return {
+                'basename': file_obj.name,  # pylint: disable=E1101
+                'dirname': str(file_obj.parent),  # pylint: disable=E1101
+                'nameroot': file_obj.stem,  # pylint: disable=E1101
+                'nameext': file_obj.suffix,  # pylint: disable=E1101
+                'checksum': file_checksum,
+                'size': size,  # pylint: disable=E1101
+                'valid': valid,
+            }
+        except (FileNotFoundError, ValueError):
+            return None
+
+    @staticmethod
+    def reconstruct_json(data: list | str) -> str | dict:
+        """Reconstruct JSON structure from provided data.
+
+        Args:
+            data (list | str): A tuple of (OutputFileInternal, json_structure) if file_id is not None,
+                else just the output string based on the analysis_outputs table.
+
+        Returns:
+            dict: Should return the JSON structure based on the input data.
+        """
+        root: dict = {}
+        if isinstance(data, str):
+            return data
+
+        def add_to_structure(current, path, content):
```

Comment:
Would be good to have some more types here, and probably some comments. It is pretty tricky to tell what is going on.